File: victimintents/src/main/java/com/csy6/victimintents/MainActivity.java
package com.csy6.victimintents;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        if(getIntent()!=null && getIntent().getParcelableExtra("intent") != null){
            startActivity((Intent)getIntent().getParcelableExtra("intent"));
        }
    }

    public void checkCode(View view) {
        String code = ((EditText) findViewById(R.id.ed_secretCode)).getText().toString();
        if("1234".equals(code)){
            startActivity(new Intent(this,SecretActivity.class));
            this.finish();
        } else {
            Toast.makeText(this,"wrong code!",Toast.LENGTH_SHORT).show();
        }
    }
}

File: forreverse/src/main/java/com/csr1/forreverse/MainActivity.kt
package com.csr1.forreverse

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider

class MainActivity : AppCompatActivity() {
    lateinit var mainViewModel: MainViewModel
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mainViewModel = ViewModelProvider(
                this,
                ViewModelProvider.AndroidViewModelFactory.getInstance(application)
        ).get(MainViewModel::class.java)
        mainViewModel.valueLiveData.observe(this, Observer {
            if (it == 1)
                supportFragmentManager.beginTransaction().replace(R.id.container, FirstFragment.newInstance("a", "b")).addToBackStack(null).commit()
            else {
                supportFragmentManager.beginTransaction().replace(R.id.container, SecondFragment.newInstance("a", "b")).addToBackStack(null).commit()
            }
        })
        val btn1 = findViewById<Button>(R.id.btn_1)
        btn1.setOnClickListener { mainViewModel.valueLiveData.value = 1 }
        findViewById<Button>(R.id.button2).setOnClickListener { mainViewModel.valueLiveData.value = 2 }

    }
}